pol-core/bscript/compiler/astbuilder/ValueBuilder.cpp

```diff
@@ -62,22 +62,22 @@ std::unique_ptr<IntegerValue> ValueBuilder::integer_value(
 }
 
 std::unique_ptr<StringValue> ValueBuilder::string_value(
-    antlr4::tree::TerminalNode* string_literal )
+    antlr4::tree::TerminalNode* string_literal, bool expect_quotes )
 {
   auto loc = location_for( *string_literal );
-  return std::make_unique<StringValue>( loc, unquote( string_literal ) );
+  return std::make_unique<StringValue>( loc, unquote( string_literal, expect_quotes ) );
 }
 
-std::string ValueBuilder::unquote( antlr4::tree::TerminalNode* string_literal )
+std::string ValueBuilder::unquote( antlr4::tree::TerminalNode* string_literal, bool expect_quotes )
```

Comment:
When using this in a normal string, the start and ending tokens are `"`. This is not the case with interstrings: `$"Hello!"` -> the `$"` is interstring_start, and the `H` begins the string part